packages/core/src/shared/telemetry/telemetry.ts

```diff
@@ -23,7 +23,10 @@ export const telemetry = new TelemetryTracer()
  */
 declare module './telemetry.gen' {
     interface Metric<T extends MetricBase = MetricBase> {
+        run<U>(fn: (span: Span<T>) => U, options?: SpanOptions): U
+    }
+
+    interface Span<T extends MetricBase = MetricBase> {
         increment(data: { [P in NumericKeys<T>]+?: number }): void
-        run<U>(fn: (span: this) => U, options?: SpanOptions): U
     }
```

Comment:
Why doesn't span have run()? I don't quite understand the distinction here. I think of metrics as mostly synonymous with spans.

Reply:
The problem is [this span as any is a lie](https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/shared/telemetry/spans.ts#L367). It never actually had run. When you try and do `span.run` you get "span.run is not a function"

The passed in span is just a telemetry span that has emit, record, increment but isn't a metric type